overlay.d/05core/usr/lib/dracut/modules.d/30ignition-coreos/coreos-teardown-initramfs.sh

```diff
@@ -33,14 +33,50 @@ selinux_relabel() {
     fi
 }
 
+# Add a function that will determine if the generated NM connection
+# profiles match the default that would be given to us if the user had
+# provided no additional configuration. i.e. did the user give us any
+# network configuration other than the default? We determine this by
+# comparing the generated output of nm-initrd-generator with a new run
+# of nm-initrd-generator. If it matches then it was the default, if not
+# then the user provided something extra.
+are_default_NM_configs() {
+    # Make two dirs for storing files to use in the comparison
+    mkdir -p /run/initramfsteardown/connectionscompare{1,2}
+    # Make another that's just a throwaway for the initrd-data-dir
+    mkdir -p /run/initramfsteardown/initrd-data-dir
+    # Copy over the previously generated connection(s) profiles
+    cp  /run/NetworkManager/system-connections/* /run/initramfsteardown/connectionscompare1
+    # Do a new run with the default input
+    /usr/libexec/nm-initrd-generator -c /run/initramfsteardown/connectionscompare2  \
+                                     -i /run/initramfsteardown/initrd-data-dir -- ip=dhcp,dhcp6
+    # remove unique identifiers from the files (so our diff can work)
+    sed -i '/^uuid=/d' /run/initramfsteardown/connectionscompare{1,2}/*
+    # currently the output will differ based on whether rd.neednet=1
+    # was part of the kargs. Let's ignore the single difference (wait-device-timeout)
+    sed -i '/^wait-device-timeout=/d' /run/initramfsteardown/connectionscompare{1,2}/*
+    if diff -r -q /run/initramfsteardown/connectionscompare{1,2}/; then
+        rc=0 # They are the default configs
+    else
+        rc=1 # They are not the defaults, user must have added configuration
+    fi
+    rm -f /run/initramfsteardown/connectionscompare{1,2}/*
+    rmdir /run/initramfsteardown/connectionscompare{1,2}
```

Comment:
Why not just

```suggestion
    rm -rf /run/initramfsteardown
```

?

Reply:
`rm -rf` scares me, but i'll change it.